frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -113,13 +116,22 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 	}
 	debug_assert_eq!(fn_weight.len(), methods.len());
 
-	let fn_doc = methods
-		.iter()
-		.map(|method| {
+	let map_fn_docs = if !def.dev_mode {
+		// Emit the [`Pallet::method`] documentation only for non-dev modes.
+		|method: &CallVariantDef| {
 			let reference = format!("See [`Pallet::{}`].", method.name);
 			quote!(#reference)
-		})
-		.collect::<Vec<_>>();
+		}
+	} else {
+		// For the dev-mode do not provide a documenation link as it will break the
+		// `cargo doc` if the pallet is private inside a test.
```

Comment:
Why is it trying to document a test? I think anything testing should not need to be documented anyway.

Reply:
I believe after that substrate PR, we no longer propagate `method.docs` and we do instead provide a link to `[Pallet:: method]`. This link will cause the `cargo doc` step on the cumulus on `test/runtime/src/test_pallet.rs`, even tho it is declared as:

```rust
#[frame_support::pallet(dev_mode)]
pub mod pallet {
#[pallet::call]
	impl<T: Config> Pallet<T> {
		/// A test dispatchable for setting a custom head data in `validate_block`.
		#[pallet::weight(0)]
		pub fn set_custom_validation_head_data(
		...
		
```